analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MemberInitializerRedundant.CSharp9.cs

```diff
@@ -1,13 +1,119 @@
-﻿using System.Runtime.CompilerServices;
+﻿using System;
+using System.Runtime.CompilerServices;
 
 class Person
 {
+    const int forCoverage = 0;
     static int minAge = 0;
-    static int maxAge = 42; // FN {{Remove the member initializer, module initializer sets an initial value for the member.}}
+    static int Bar { get; set; } = 1;
+    static event EventHandler Quix = (a, b) => { };
+
+    static int maxAge = 42; // Noncompliant {{Remove the static member initializer, a static constructor or module initializer sets an initial value for the member.}}
+    static string Name = "X"; // Noncompliant
+    static string Job = "1"; // Noncompliant
+    static bool Nice = true; // Noncompliant
+    static int Foo { get; set; } = 1; // Noncompliant
+    static event EventHandler Taz = (a, b) => { }; // Noncompliant
+
+    public Person() { } // for coverage
 
     [ModuleInitializer]
     internal static void Initialize()
     {
         maxAge = 42;
     }
+
+    [Obsolete]
+    [ModuleInitializer]
+    internal static void SecondInitialize()
+    {
+        Name = "Foo";
+        Job = "2";
+        Nice = false;
+        Foo = 1;
+        Taz = (a, b) => { };
+    }
+
+    [ModuleInitializer]
+    internal static void ThirdInitialize()
+    {
+        Job = "3";
+    }
+
+    [Obsolete]
+    internal static void Foo1()
+    {
+        minAge = 2;
+    }
+
+    [Obsolete("argument", false)]
+    internal static void Foo2()
+    {
+        minAge = 2;
+    }
+
+    internal static void Foo3()
+    {
+        minAge = 3;
+        Bar = 2;
+        Quix = (a, b) => { };
+    }
+}
+
+// we do not support records for instance members, see discussion in https://github.com/SonarSource/sonar-dotnet/pull/4756
+record Foo
+{
+    static int foo = 1; // FN
+    int bar = 42; // FN, reported by the C# compiler
+    int fred { get; init; } = 42; // FN, reported by the C# compiler
```

Comment:
Is it really a FN? I would say we intentionally took the decision not to raise as the compiler reports the issue.

Reply:
I retracted my comments - I am not sure what's enabled in sharplab.io - it seems they have the messages from the IDE enabled (e.g. unused using), which don't necessarily come with the compiler. Also, I couldn't get this behavior with my local IDE - it may be on the preview version. So removed those comments.

The main point is that I need to cut a branch regarding records because:
- in sharplab.io, there's already a message, which suggests it's either in the IDE or in a future compiler version - I need to double check
- it's more complex due to how record constructors get compiled into intermediate C# code

Reply:
Maybe it deserves a background task or an issue, to investigate what is the exact state with the records.

Reply:
I opened https://trello.com/c/ZMEesp9Q/279-c-roslyn-new-vs-analyzers-for-records-and-s3604